Add release process instructions (#37)
rivet-gg/plugin-godot: README.md

<p align="center">
    <picture>
        <source media="(prefers-color-scheme: dark)" srcset="./addons/rivet/images/icon-text-white.svg">
        <img src="./addons/rivet/images/icon-text-black.svg">
    </picture>
</p>
<h1 align="center">Rivet Godot Plugin</h1>
<p align="center">
    <a href="https://rivet.gg/discord"><img src="https://img.shields.io/discord/822914074136018994"></a>
</p>

---

> [!WARNING]
> _Godot 3 is not supported at the moment._

## 📦 Installation

### Using [`gd-plug`](https://github.com/imjp94/gd-plug)

1. Add this line to your `plug.gd`:
```gdscript
plug("rivet/plugin-godot")
```

> 
> 

Reviewer: Nit: markdown formatting
```suggestion

```

> ### Manual installation
> 1. [Download the latest release](https://github.com/rivet-gg/plugin-godot/releases/latest)

Reviewer: nit: markdown formatting

```suggestion
### Manual installation

1. [Download the latest release](https://github.com/rivet-gg/plugin-godot/releases/latest)
```

2. Unpack the `addons/rivet` folder into your `/addons` folder within the Godot project
3. Enable this addon within the Godot settings `Project > Project Settings > Plugins`

## 🚀 Getting started

> [Head over to our Learning Center to get started with Rivet](https://rivet.gg/learn/godot).

Reviewer: I prefer only linking part of the sentence:

```suggestion
Head over to our [Learning Center](https://rivet.gg/learn/godot) to get started with Rivet and Godot.
```

## 🧐 Features

### Link your game with Rivet

> One click and it's done! Linking your game with Rivet lets you change the configuration of the deployed game server, set matchmaker options, and much more!

Reviewer: Can we include an image in the repo that shows the screens for each of these points?

Also, as I'm writing the docs, I'm finding that it might be best to start with an approach that doesn't describe the plugin with having "endless possibilities". I might reword this as such:

```suggestion
One click and your game is connected! Linking your game with Rivet gives you one-click deployments to production, easy playtesting setup, and ways to configure your game's hosting and matchmaking.
```

(I don't think this should be the final version, I didn't figure out the last sentence the best)
